OptimizelySDK/DatafileManagement/ProjectConfig.cs

```diff
@@ -0,0 +1,269 @@
+﻿/* 
+ * Copyright 2019, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using OptimizelySDK.Entity;
+using System.Collections.Generic;
+
+namespace OptimizelySDK
+{
+    public interface ProjectConfig
```

Comment:
ProjectConfig can be more abstract than this. We can have project configs later on that are not based on datafile, but can be based on a different structure. Here is the abstraction approach we took in the Java SDK: 

https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/config/ProjectConfig.java

For this reason, also, we should change the name of the package to `ProjectConfig` instead of `DatafileManagement` so we don't tie it to just the Datafile transport type.